packages/cozy-clisk/src/contentscript/ContentScript.js

```diff
@@ -164,7 +164,27 @@ export default class ContentScript {
     log.info(`I am the ${contentScriptType}`)
 
     if (contentScriptType === WORKER_TYPE) {
-      this.onWorkerReady()
+      // wait at most 1s for the body to be present in the page, this is for compatibility between
+      // iphone and android. onWorkerReady was called too soon on iphone and the content script
+      // could not subscribe to page events. And we don't want to wait for document.body in every
+      // konnectors.
+      let found = true
+      await waitFor(() => Boolean(document.body), {
+        interval: 10,
+        timeout: {
+          milliseconds: 1000,
+          fallback: () => {
+            found = false
+            log.warn(
+              'Could not find any body in the current page after 1s',
+              window.location.href
+            )
+          }
+        }
+      })
+      if (found) {
+        this.onWorkerReady()
+      }
```

Comment:
Can't you use https://github.com/react-native-webview/react-native-webview/blob/master/docs/Reference.md#onloadend instead of relying on that? Adding a completely random timeout is not great.

Reply:
From my tests, at least on android, it looks like onLoadEnd is called before the init of the worker, then at this step it is impossible to call the worker to send it a message.